### ISSUE TITLE
Stash pop removes stash from list even when there are conflicts

### ISSUE THREAD

**Issue by fmorroni** [gitui-org/gitui#2372]
**Describe the bug**
Stashes are removed from the stash list when popped even if there are conflicts. This is different from the git standard. From [git stash documentation](https://git-scm.com/docs/git-stash#Documentation/git-stash.txt-pop--index-q--quietltstashgt):

> pop [--index] [-q|--quiet] [\<stash>]
> ... Applying the state can fail with conflicts; in this case, **it is not removed from the stash list**.

**To Reproduce**
1. Stash some changes.
3. Make some changes to a file that will cause conflicts with the stash.
4. Commit the changes.
5. Pop the stash.
6. Stash is applied and removed from the list.

**Expected behavior**
Stash should be applied but not removed in case of conflicts.

**Context:**
 - OS: `Arch Linux x86_64` - Kernel: `6.10.10-arch1-1`
 - gitui --version: `gitui nightly 2024-09-12 (0f5cf89)`
 - pacman -Qi gitui: `0.26.3-2`
 - Rust version: `1.81.0`

**Comment by wugeer** [comment]
@fmorroni I can't reproduce your issue, there is something wrong?
Here are the steps I followed.
[录屏 2024-09-30 09-26-37.webm](https://github.com/user-attachments/assets/e6e37f7a-a29e-405f-91a8-1be1dc3d68a6)
![image](https://github.com/user-attachments/assets/667fb9fa-031f-4247-a9aa-ffe16eb455ea)

**Comment by fmorroni** [comment]
@wugeer my bad, the instructions were wrong. You should actually commit the changes that cause the conflict and then pop the stash.

**Comment by wugeer** [comment]
@fmorroni Indeed, the issue you mentioned is occurring here. I will try to resolve it. :)
![image](https://github.com/user-attachments/assets/8aabe5ef-626a-47a0-b5ed-8edff955738d)
